src/index.ts

```diff
@@ -1,107 +1,25 @@
 import isSymbol from 'is-symbol';
 import isObjectAny from 'isobject';
 import { get } from 'lodash-es';
-import memoize from 'memoizerific';
 import { extractEventHiddenProperties } from './dom-event';
 
-// eslint-disable-next-line @typescript-eslint/ban-types, no-use-before-define
 const isObject = isObjectAny as <T = object>(val: any) => val is T;
 
-const removeCodeComments = (code: string) => {
-  let inQuoteChar = null;
-  let inBlockComment = false;
-  let inLineComment = false;
-  let inRegexLiteral = false;
-  let newCode = '';
-
-  if (code.indexOf('//') >= 0 || code.indexOf('/*') >= 0) {
-    for (let i = 0; i < code.length; i += 1) {
-      if (!inQuoteChar && !inBlockComment && !inLineComment && !inRegexLiteral) {
-        if (code[i] === '"' || code[i] === "'" || code[i] === '`') {
-          inQuoteChar = code[i];
-        } else if (code[i] === '/' && code[i + 1] === '*') {
-          inBlockComment = true;
-        } else if (code[i] === '/' && code[i + 1] === '/') {
-          inLineComment = true;
-        } else if (code[i] === '/' && code[i + 1] !== '/') {
-          inRegexLiteral = true;
-        }
-      } else {
-        if (
-          inQuoteChar &&
-          ((code[i] === inQuoteChar && code[i - 1] !== '\\') ||
-            (code[i] === '\n' && inQuoteChar !== '`'))
-        ) {
-          inQuoteChar = null;
-        }
-        if (inRegexLiteral && ((code[i] === '/' && code[i - 1] !== '\\') || code[i] === '\n')) {
-          inRegexLiteral = false;
-        }
-        if (inBlockComment && code[i - 1] === '/' && code[i - 2] === '*') {
-          inBlockComment = false;
-        }
-        if (inLineComment && code[i] === '\n') {
-          inLineComment = false;
-        }
-      }
-      if (!inBlockComment && !inLineComment) {
-        newCode += code[i];
-      }
-    }
-  } else {
-    newCode = code;
-  }
-
-  return newCode;
-};
-
-const cleanCode = memoize(10000)((code: string) =>
-  removeCodeComments(code)
-    .replace(/\n\s*/g, '') // remove indents & newlines
-    .trim()
-);
-
-const convertShorthandMethods = function convertShorthandMethods(key: string, stringified: string) {
-  const fnHead = stringified.slice(0, stringified.indexOf('{'));
-  const fnBody = stringified.slice(stringified.indexOf('{'));
-
-  if (fnHead.includes('=>')) {
-    // This is an arrow function
-    return stringified;
-  }
-
-  if (fnHead.includes('function')) {
-    // This is an anonymous function
-    return stringified;
-  }
-
-  let modifiedHead = fnHead;
-
-  modifiedHead = modifiedHead.replace(key, 'function');
-
-  return modifiedHead + fnBody;
-};
-
 const dateFormat = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d{3})?Z$/;
 
 export interface Options {
   allowRegExp: boolean;
-  allowFunction: boolean;
   allowSymbol: boolean;
   allowDate: boolean;
   allowUndefined: boolean;
-  allowClass: boolean;
   allowError: boolean;
   maxDepth: number;
   space: number | undefined;
   lazyEval: boolean;
```

Comment:
Can be removed too?!

Reply:
It needs eval for full support.
I've changed the implementation to make it a plain object.